.github/workflows/release.yml

```diff
@@ -0,0 +1,106 @@
+name: Release new app version
+on:
+  push:
+    tags: /^[0-9]+\.[0-9]+\.[0-9]+(-rc\.[0-9]+)?$/
+jobs:
+  run-static-checks:
+     uses: ./.github/workflows/staticcheck.yaml
+  release-android:
+    needs: run-static-checks
+    environment: prod
+    runs-on: ubuntu-latest
+    steps:
+      - id: checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab #v3.5.2
+      - id: setup
+        uses: ./.github/actions/setup-composite
+      - id: setup-jdk-11
+        uses: actions/setup-java@5ffc13f4174014e2d4d4572b3d74c3fa61aeb2c2 #v3.11.0
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: gradle
+      - id: setup-android-sdk
+        uses: android-actions/setup-android@7c5672355aaa8fde5f97a91aa9a99616d1ace6bc #v2.0.10
+      - id: setup-ruby
+        uses: ruby/setup-ruby@d2b39ad0b52eca07d23f3aa14fdf2a3fcc1f411c #v1.149.0
+        with:
+          ruby-version: 2.7
+          bundler-cache: true
+      - id: build-release-android
+        run: |
+          ./scripts/android-release.sh ./android/app
+          yarn bundle:android-release
+          cd android && bundle exec fastlane alpha
+          cd android && bundle exec fastlane promote_internal_to_alpha
+        shell: bash
+        env:
+          ENCODED_IOAPP_GOOGLE_SERVICES_JSON_FILE: ${{secrets.ENCODED_IOAPP_GOOGLE_SERVICES_JSON_FILE}}
+          ENCODED_IOAPP_JSON_KEY_FILE: ${{secrets.ENCODED_IOAPP_JSON_KEY_FILE}}
+          ENCODED_IO_APP_RELEASE_KEYSTORE: ${{secrets.ENCODED_IO_APP_RELEASE_KEYSTORE}}
+          IO_APP_RELEASE_STORE_FILE : ${{secrets.IO_APP_RELEASE_STORE_FILE}}
+          IO_APP_RELEASE_STORE_PASSWORD: ${{secrets.IO_APP_RELEASE_STORE_PASSWORD}}
+          IO_APP_RELEASE_KEY_ALIAS: ${{secrets.IO_APP_RELEASE_KEY_ALIAS}}
+          IO_APP_RELEASE_KEY_PASSWORD: ${{secrets.IO_APP_RELEASE_KEY_PASSWORD}}
+      # Store artifacts
+      - id: archive-production-artifacts
+        uses: actions/upload-artifact@v3
```

Comment:
Here we're tagging instead of pinning ;)

Reply:
Actually we need to remove this we can't upload the artifact of a build unfortunately in a public repo